### PR TITLE
Cancel previous ongoing runs in the PR/branch when you push new commits 

### LINKED ISSUE #4781: Cancel previous ongoing runs in the PR/branch when you push new commits
**Summary:** 
It makes sense to only build the latest commit and cancel any previous ongoing builds when you push new commits to a PR or branch. This will ensure that we don't run out of free build mins.

Link for help https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre

**Would you like to work on the issue?**
No

### PR DESCRIPTION
**Pull request by RishiKumarRay**
**Description (required)**

Fixes #4781 

What changes did you make?
Adding this to cancel any previous ongoing builds when you push new commits to a PR or branch. This will ensure that we don't run out of free build mins.

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -12,10 +12,16 @@ jobs:
       uses: actions/setup-java@v1
       with:
         java-version: 1.8
+        
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.8.0
```
Comment (nicolas-raoul) [inline .github/workflows/android.yml:17]: [Latest version seems to be 0.9.1](https://github.com/marketplace/actions/cancel-workflow-action), any reason to not use that one?
If yes, would you mind adding a comment?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: .github/workflows/android.yml (hunk 1 above)
Comment (RishiKumarRay) [reply .github/workflows/android.yml]: @nicolas-raoul sorry i missed out that, can you review now ? i have updated

**Review by madhurgupta10 — CHANGES_REQUESTED** [review]
@RishiKumarRay consider using concurrency, example https://github.com/TeamAmaze/AmazeFileManager/blob/release/3.7/.github/workflows/android-feature.yml

**Comment by RishiKumarRay** [comment]
@madhurgupta10 can you check now ?

**Comment by madhurgupta10** [comment]
@RishiKumarRay Now revert the extra changes, this will also test if ongoing builds are cancelled or not

**Comment by RishiKumarRay** [comment]
@madhurgupta10 how to revert?

**Comment by madhurgupta10** [comment]
> @madhurgupta10 how to revert?

You can push another commit by removing the unnecessary changes

**Comment by RishiKumarRay** [comment]
@madhurgupta10 there was only some extra spaces i have removed that please check

**Review by madhurgupta10 — APPROVED** [review]
LGTM, happy to merge after build completes

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -11,11 +15,11 @@ jobs:
     - name: set up JDK 1.8
       uses: actions/setup-java@v1
       with:
-        java-version: 1.8
+        java-version: 1.8
```
Comment (nicolas-raoul) [inline .github/workflows/android.yml:18]: For next time: Better avoid such unrelated changes.

**Comment by RishiKumarRay** [comment]
@nicolas-raoul yeah sure i will remember this. Thanks for helping out and merging @madhurgupta10
